jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/WithMavenStepExecution.java

```diff
@@ -668,25 +696,52 @@ private String setupSettingFile(@Nonnull Collection<Credentials> credentials) th
             FilePath settings;
             if ((settings = ws.child(settingsPath)).exists()) {
                 // Settings file residing on the agent
-                console.format("[withMaven] use Maven settings provided by the Jenkins global configuration on the build agent '%s' %n", settingsPath);
                 settings.copyTo(settingsDest);
+                envOverride.put("MVN_SETTINGS", settingsDest.getRemote());
+                if (LOGGER.isLoggable(Level.FINE)) {
+                    mavenSettingsLog.append("Maven settings on the build agent'").append(settingsPath).append("'");
+                    console.println(mavenSettingsLog);
+                }
+                return settingsDest.getRemote();
             } else {
                 throw new AbortException("Could not find file provided by the Jenkins global configuration '" + settings + "' on the build agent");
             }
-            envOverride.put("MVN_SETTINGS", settingsDest.getRemote());
-            return settingsDest.getRemote();
+
         } else if (settingsProvider instanceof DefaultSettingsProvider) {
             // do nothing
+            if (LOGGER.isLoggable(Level.FINE)) {
+                mavenSettingsLog.append("Maven settings defined by 'DefaultSettingsProvider', NOT overriding it.");
+                console.println(mavenSettingsLog);
+            }
         } else if (settingsProvider == null) {
             // should not happen according to the source code of jenkins.mvn.MavenConfig.getSettingsProvider() in jenkins-core 2.7
             // do nothing
+            if (LOGGER.isLoggable(Level.FINE)) {
+                mavenSettingsLog.append("Maven settings are null. NO settings will be defined.");
+                console.println(mavenSettingsLog);
+            }
         } else {
             console.println("[withMaven] Ignore unsupported Maven SettingsProvider " + settingsProvider);
         }
 
         return null;
     }
 
+    @CheckForNull
+    private MavenConfigFolderOverrideProperty getMavenConfigOverrideProperty() {
+        Job<?, ?> job = build.getParent();
+        ItemGroup<?> group = job.getParent();
+        while (group != null && !(group instanceof Jenkins) && group instanceof AbstractFolder) {

```

Comment:
not sure but can't you have sth like this hierarchy:

AbstractFolder
   someOtherGroupType
       Job


for which you still want to read the abstract folder mvn settings?

Reply:
hummm, strange but I guess it is possible, I haven't seen a group not being an AbstractFolder, other than jenkins it self. Let me revisit that code.